Metric properties of real negation, absolute value, addition and maximum (#1398)
UniMath/agda-unimath: src/real-numbers/isometry-negation-real-numbers.lagda.md

Reviewer: Could you add a link to this file in the "see also" section of "negation of real numbers," stating something like

````md
## See also

- In [The negation isometry on real numbers](real-numbers.isometry-negation-real-numbers.md) we show that negation is an [isometry](metric-spaces.isometries-metric-spaces.md) on the [metric space of real numbers](real-numbers.metric-space-of-real-numbers.md)
````

> # The isometric negation of real numbers

Reviewer: ```suggestion
# The negation isometry on real numbers
```

open import metric-spaces.isometries-metric-spaces
open import metric-spaces.metric-spaces

open import real-numbers.dedekind-real-numbers
open import real-numbers.inequality-real-numbers
open import real-numbers.metric-space-of-real-numbers
open import real-numbers.negation-real-numbers
```

</details>

## Idea

[Negation](real-numbers.negation-real-numbers.md) of real numbers is an
[isometry](metric-spaces.isometries-metric-spaces.md) of the
[metric space of real numbers](real-numbers.metric-space-of-real-numbers.md).

## Definitions

### Negation of a real number preserves neighborhoods

```agda
module _
  {l1 : Level}
  where

  neg-neighborhood-leq-ℝ : (d : ℚ⁺) (x y : ℝ l1) →
    is-in-neighborhood-leq-ℝ l1 d x y →
    is-in-neighborhood-leq-ℝ l1 d (neg-ℝ x) (neg-ℝ y)
  neg-neighborhood-leq-ℝ d x y H =
    neighborhood-real-bound-each-leq-ℝ
      ( d)
      ( neg-ℝ x)
      ( neg-ℝ y)
      ( reverses-lower-neighborhood-leq-neg-ℝ
        ( d)
        ( y)
        ( x)
        ( right-real-bound-neighborhood-leq-ℝ d x y H))
      ( reverses-lower-neighborhood-leq-neg-ℝ
        ( d)
        ( x)
        ( y)
        ( left-real-bound-neighborhood-leq-ℝ d x y H))
```

> ### Negation a real number is an isometry `ℝ → ℝ`

Reviewer: ```suggestion
### Negation on the real numbers is an isometry
```
